add documentation for spaceport (#445)
apollographql/router: docs/source/configuration.mdx

                  - value1
                  - value2
```

## Spaceport

Spaceport reports usage data to the Apollo ingress. You can see how it is configured in the examples above.

> NB: This component is still evolving and you should expect functionality to change and evolve here. We are actively seeking user feedback as we work to shape this component.

Reviewer: ```suggestion
> Note: This component is still evolving and you should expect functionality to change and evolve here. We are actively seeking user feedback as we work to shape this component.
```

### No usage reporting

If you don't wish to take advantage of usage reporting, then no configuration is required. This is the default.

### Usage reporting

```yaml:title=configuration.yaml
# Apollo Studio Graph configuration details
# If we don't specify a graph section, then no statistics are collected for
# processing, since there is no point collecting statistics unless we have
# somewhere to send them.
graph:
  # Which graph should our Spaceport be sending data about?
  reference: Benchmarking@current
  # Key used to send data
  key: service:<redacted>
```

> You must configure the graph section. If you specify a graph section, then both the reference and the key must be specified. The reference must refer to a valid graph/version combination against which statistics are reported. The Key must be a valid API key which can be used for reporting statistics.

Reviewer: ```suggestion
You must configure the `graph` section. If you specify a `graph` section, then both the `reference` and the `key` must be specified. The `reference` must refer to a valid graph/variant combination (i.e., the "graph ref") against which statistics are reported. The `key` must be a valid Apollo Studio API key which can be used for reporting statistics (a [Studio Graph _Contributor_ token](https://www.apollographql.com/docs/studio/org/members/#organization-wide-member-roles) is appropriate).
```

For more details on this process, refer to the [usage reporting documentation](https://www.apollographql.com/docs/studio/metrics/usage-reporting/)

> The key fulfills the same function as the APOLLO_KEY environment variable.

Reviewer: ```suggestion
The key fulfills the same function as the `APOLLO_KEY` environment variable.
```

> The reference fulfills the same function as the APOLLO_GRAPH_REF environment variable.

Reviewer: ```suggestion
The `reference` fulfills the same function as the `APOLLO_GRAPH_REF` environment variable.
```

> NB: These environment variables are *not* currently recognised by the router. This is being tracked by [issue](https://github.com/apollographql/router/issues/425)

Reviewer: ```suggestion
> Note: These environment variables are *not* currently recognised by the router. This is being tracked by [#425](https://github.com/apollographql/router/issues/425)
```

> ### Spaceport Configuration

Reviewer: ```suggestion
### Spaceport configuration
```

The spaceport can either be run as an internal, router specific component or as an external resource shared by multiple routers.

```yaml:title=configuration.yaml
# Spaceport configuration. These values are the default values if not specified
spaceport:
  # Should we spawn a Spaceport or talk to an existing external Spaceport?
  external: false
  # What's the address of the Spaceport?
  collector: https://127.0.0.1:50051
  # Where should we listen IF we have spawned an internal Spaceport?
  listener: 0.0.0.0:50051
```

> #### Internal Spaceport

Reviewer: ```suggestion
#### Internal spaceport
```

This is the default and is most appropriate for evaluations, demos, development support and generally getting started. This requires no configuration by default and will be automatically started if the graph configuration is present.
In this mode it is possible to further configure the address at which the internal spaceport is listening and the address of the spaceport to use. This should only be necessary if there is a clash on the default port, for instance if running multiple routers on the same system or clashing with another application.

> #### External Spaceport

Reviewer: ```suggestion
#### External spaceport
```
